client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/HotRodAsyncCaches.java

```diff
@@ -0,0 +1,72 @@
+package org.infinispan.client.hotrod;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.api.async.AsyncCache;
+import org.infinispan.api.async.AsyncCaches;
+import org.infinispan.api.configuration.CacheConfiguration;
+import org.infinispan.client.hotrod.impl.InternalRemoteCache;
+import org.infinispan.commons.util.concurrent.CompletableFutures;
+
+/**
+ * @since 14.0
+ **/
+final class HotRodAsyncCaches implements AsyncCaches {
+   private final HotRod hotrod;
+
+   HotRodAsyncCaches(HotRod hotrod) {
+      this.hotrod = hotrod;
+   }
+
+   @Override
+   public <K, V> CompletionStage<AsyncCache<K, V>> get(String name) {
+      InternalRemoteCache<K, V> cache = getCache(name);
+      return CompletableFuture.completedFuture(new HotRodAsyncCache<>(hotrod, cache));
+   }
+
+   @Override
+   public <K, V> CompletionStage<AsyncCache<K, V>> create(String name, CacheConfiguration cacheConfiguration) {
+      hotrod.cacheManager.getConfiguration().addRemoteCache(name, builder -> builder.configuration(cacheConfiguration.toString()));
+      return get(name);
+   }
+
+   @Override
+   public <K, V> CompletionStage<AsyncCache<K, V>> create(String name, String template) {
+      hotrod.cacheManager.getConfiguration().addRemoteCache(name, builder -> builder.templateName(template));
+      return get(name);
+   }
+
+   @Override
+   public CompletionStage<Void> remove(String name) {
+      hotrod.cacheManager.administration().removeCache(name);
```

Comment:
Sadly all of these methods are really non blocking.. but they are just doing an await call.

Reply:
Yeah. I had another branch change the RemoteCacheManagerImpl to return a CompletableFuture, but I didn't pull it into this PR so the change wouldn't be lost in the huge diff.